go/state/state_db_test.go

```diff
@@ -4143,6 +4143,71 @@ func TestSlotIdOrder(t *testing.T) {
 	}
 }
 
+func TestStateDB_SetTransientState_RollbackRemovesValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	s := db.Snapshot()
+	db.SetTransientState(address1, key1, val1)
+
+	db.RevertToSnapshot(s)
+	if got, want := db.GetTransientState(address1, key1), val0; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+}
+
+func TestStateDB_SetTransientState_RollbackToPreviousValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	db.SetTransientState(address1, key1, val1)
+	s := db.Snapshot()
+	// Save previous value
+	want := db.GetTransientState(address1, key1)
+
+	db.SetTransientState(address1, key1, val2)
```

Comment:
same as above, you should test here that the value has really changed before it was reverted